es/es-server/src/main/java/org/elasticsearch/index/engine/Engine.java

```diff
@@ -1368,19 +1373,38 @@ public long getIfPrimaryTerm() {
     }
 
     public static class GetResult implements Releasable {
-
+        private final boolean exists;
+        private final long version;
         @Nullable
         private final DocIdAndVersion docIdAndVersion;
-        private final Searcher searcher;
+        private final Engine.Searcher searcher;
 
-        static final GetResult NOT_EXISTS = new GetResult(null, null);
+        public static final GetResult NOT_EXISTS = new GetResult(false, Versions.NOT_FOUND, null, null);
+
+        private GetResult(boolean exists, long version, DocIdAndVersion docIdAndVersion, Engine.Searcher searcher) {
+            this.exists = exists;
+            this.version = version;
+            this.docIdAndVersion = docIdAndVersion;
+            this.searcher = searcher;
+        }
 
         /**
          * Build a non-realtime get result from the searcher.
          */
-        GetResult(Searcher searcher, DocIdAndVersion docIdAndVersion) {
-            this.searcher = searcher;
-            this.docIdAndVersion = docIdAndVersion;
+        public GetResult(Engine.Searcher searcher, DocIdAndVersion docIdAndVersion) {
+            this(true, docIdAndVersion.version, docIdAndVersion, searcher);
+        }
+
+        public boolean exists() {
+            return exists;
+        }
+
+        public long version() {
+            return this.version;
+        }
```

Comment:
Can we remove these extra properties again? See https://github.com/crate/crate/commit/836b1bb176b3c1ce4116e2d812597b37da8a6740 for why they were removed.